pkg/services/ngalert/models/alert_rule.go

```diff
@@ -294,6 +294,11 @@ type AlertRule struct {
 	IsPaused             bool
 	NotificationSettings []NotificationSettings
 	Metadata             AlertRuleMetadata
+	// MissingSeriesEvalsToResolve specifies the number of consecutive evaluation intervals
+	// required before resolving an alert state (a dimension) when data is missing.
+	// If nil, alerts resolve after 2 missing evaluation intervals
+	// (i.e., resolution occurs during the second evaluation where data is absent).
+	MissingSeriesEvalsToResolve *int
```

Comment:
What do you think about making it even more flexible and let user define `reason`. This could be useful in the case of Prometheus rules where threshold is in the query and, therefore, reason `Missing series` does not make any sense.

Reply:
I agree, this will be useful especially for Prometheus rules. There will be more PRs! :)